src/Timeline.js

```diff
@@ -179,27 +177,86 @@ export default class Timeline extends Axis {
     this._brushGroup.selectAll(".overlay")
       .attr("cursor", this._brushing ? "crosshair" : "pointer");
 
-    this._brushGroup.selectAll(".selection")
+    const brushSelection = this._brushGroup.selectAll(".selection")
       .call(attrize, this._selectionConfig)
       .attr("height", timelineHeight);
 
-    this._brushGroup.selectAll(".handle")
+    const brushHandle = this._brushGroup.selectAll(".handle")
       .call(attrize, this._handleConfig)
-      .attr("height", timelineHeight + this._handleSize);
+      .attr("height", this._buttonBehaviorCurrent === "buttons" ? this._buttonHeight : timelineHeight + this._handleSize);
+
+    if (this._buttonBehaviorCurrent === "buttons") {
+      const yTransform = this._height / 2 - this._buttonHeight / 2;
+
+      brushSelection.attr("y", yTransform);
+      brushHandle.attr("y", yTransform);
+    }
 
   }
 
+  /**
+      @memberof Timeline
+      @desc Update limits of the brush.
+      @private
+  */
+  _updateBrushLimit(selection) {
+    if (selection[0] === selection[1]) {
+      selection[0] -= 0.1;
+      selection[1] += 0.1;
+    }
+
+    if (this._buttonBehaviorCurrent === "buttons") {
+      const buttonWidth = 0.5 * (this._width / this._availableTicks.length - this._handleSize);
+      selection[0] -= buttonWidth;
+      selection[1] += buttonWidth;
+    }
+
+    return selection;
+  }
+
   /**
       @memberof Timeline
       @desc Draws the timeline.
       @param {Function} [*callback* = undefined]
       @chainable
   */
   render(callback) {
+    const {height, y} = this._position;
+    let ticksWidth = 0;
 
-    super.render(callback);
+    if (this._ticks) {
```

Comment:
This should be `if (this._ticks && !this._labels) {`, we don't want to override the user's labels if they have specified them.